bin/templates/cordova/lib/device.js

```diff
@@ -96,10 +96,15 @@ module.exports.install = function(target, buildResults) {
     }).then(function(resolvedTarget) {
         var apk_path = build.findBestApkForArchitecture(buildResults, resolvedTarget.arch);
         var launchName = appinfo.getActivityName();
+        var pkgName = appinfo.getPackageName();
         console.log('Using apk: ' + apk_path);
-        console.log('Installing app on device...');
-        var cmd = 'adb -s ' + resolvedTarget.target + ' install -r "' + apk_path + '"';
-        return exec(cmd, os.tmpdir())
+        console.log('Uninstalling ' + pkgName + ' from device...');
+        return exec('adb -s ' + resolvedTarget.target + ' uninstall ' + pkgName, os.tmpdir())
```

Comment:
No. Even if `adb uninstall` fails to uninstall an app, it exits with error code 0, and thus `exec` promise is always resolved in this case.